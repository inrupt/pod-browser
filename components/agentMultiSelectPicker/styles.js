/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles } from "@solid/lit-prism-patterns";

const TABLE_MAX_HEIGHT = 460;
const MOBILE_TABLE_MIN_HEIGHT = 300;

export default function styles(theme) {
  return createStyles(theme, ["icons", "button", "table"], {
    "agent-multi-select-picker": {
      minHeight: MOBILE_TABLE_MIN_HEIGHT,
      display: "flex",
      flexDirection: "column",
      backgroundColor: theme.palette.common.white,
      borderRadius: theme.shape.borderRadius,
      border: `1px solid ${theme.palette.grey.A100}`,
      marginBottom: theme.spacing(2.4),
      height: "100%",
      maxHeight: TABLE_MAX_HEIGHT,
      overflowY: "auto",
      [theme.breakpoints.up("xs")]: {
        minHeight: "unset",
      },
    },
    "agent-multi-select-picker__tabs-and-button-container": {
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      borderBottom: `1px solid ${theme.palette.grey.A100}`,
    },
    "agent-multi-select-picker__agent-table": {
      width: `calc(100% - ${theme.spacing(1.6) * 2}px)`,
      boxSizing: "border-box",
      marginLeft: theme.spacing(1.6),
      marginRight: theme.spacing(1.6),
      "&& thead > tr": {
        minWidth: "100%",
        fontSize: theme.typography.caption.fontSize,
        alignItems: "center",
        borderBottom: `1px solid ${theme.palette.grey.A100}`,
        fontWeight: theme.typography.body1.fontWeight,
        color: theme.palette.text.primary,
      },
      "&& tbody tr": {
        borderBottom: `1px solid ${theme.palette.grey.A100}`,
      },
      "&& tbody td": {
        maxWidth: "100%",
      },
      "&& tbody :last-child": {
        borderBottom: "none",
      },
    },
    "agent-multi-select-picker__table-header": {
      display: "flex",
      margin: 0,
      padding: theme.spacing(0.5),
      textTransform: "capitalize",
    },
    "agent-multi-select-picker__tabs": {
      borderBottom: "none !important", // overriding the default border bottom so we can set it to the whole width of the container
    },
    "agent-multi-select-picker__empty": {
      display: "flex",
      flexDirection: "column",
      padding: theme.spacing(0.7, 1.4),
      textAlign: "center",
    },
    // hopefully we can remove these once we figure out how to test the Hidden component from MUI so we can use that instead
    "agent-multi-select-picker__mobile-only": {
      display: "flex",
      [theme.breakpoints.up("sm")]: {
        display: "none",
      },
    },
    "agent-multi-select-picker__desktop-only": {
      display: "none",
      [theme.breakpoints.up("sm")]: {
        display: "flex",
      },
    },
  });
}
