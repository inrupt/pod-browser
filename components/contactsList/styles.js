/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles, table } from "@solid/lit-prism-patterns";

const styles = (theme) => {
  const tableStyles = table.styles(theme);
  return createStyles(theme, ["container", "table", "icons"], {
    table: {
      "& tbody td": {
        "&:first-child": tableStyles["table__body-cell--width-preview"],
      },
    },
    "container-view": {
      marginTop: theme.spacing(1),
    },
    "container-view--menu-open": {
      [theme.breakpoints.up("sm")]: {
        paddingRight: "50%",
      },
      [theme.breakpoints.up("md")]: {
        paddingRight: "33.33%",
      },
      [theme.breakpoints.up("lg")]: {
        paddingRight: "25%",
      },
    },
    "container-menu": {
      display: "flex",
      justifyContent: "space-between",
      margin: theme.spacing(1, 0, 0),
    },
    avatar: {
      width: "30px",
      height: "30px",
    },
    addNewContact: {
      textAlign: "right",
    },
  });
};

export default styles;
