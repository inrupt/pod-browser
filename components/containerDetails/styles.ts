/* eslint-disable @typescript-eslint/no-explicit-any */

const styles: Record<string, any> = {
  centeredSection: {
    padding: "1rem",
  },
  raw: {
    height: "100%",
    width: "100%",
    maxHeight: "200px",
    overflow: "auto",
  },
  listItem: {
    display: "flex",
    justifyContent: "space-between",
  },
  detailText: {
    fontSize: "0.75rem",
  },
  typeValue: {
    marginLeft: "auto",
  },
};

export default styles;
