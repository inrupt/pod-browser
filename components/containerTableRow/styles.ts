/* eslint-disable @typescript-eslint/no-explicit-any */

const styles: Record<string, any> = {
  tableRow: {
    cursor: "pointer",
  },
  spinnerContainer: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "nowrap",
    justifyContent: "center",
  },
};

export default styles;
