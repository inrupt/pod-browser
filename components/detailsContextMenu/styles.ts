/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
import { PrismTheme } from "@solid/lit-prism-patterns";

const styles = (theme: PrismTheme) => ({
  "drawer__close-button": {
    width: 50,
    margin: "8px 0 0 -25px",
    backgroundColor: theme.palette.background.default,
    boxShadow: "1px 1px 3px #333",
    "&:hover": {
      backgroundColor: "#fff",
    },
  },
  "drawer__paper": {
    borderLeftStyle: "solid",
    borderLeftWidth: theme.spacing(1),
    borderLeftColor: theme.palette.grey["50"],
    width: "95%",
    [theme.breakpoints.up("sm")]: {
      width: "50%",
    },
    [theme.breakpoints.up("md")]: {
      width: "33.33%",
    },
    [theme.breakpoints.up("lg")]: {
      width: "25%",
    }
  },
});

export default styles;
