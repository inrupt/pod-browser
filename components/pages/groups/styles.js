/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

export default function styles(theme) {
  return {
    "groups-container": {
      flexGrow: 1,
      margin: theme.spacing(1, 0),
      [theme.breakpoints.up("sm")]: {
        display: "flex",
        gap: "1rem",
      },
    },
    "groups-container__content": {
      display: "none",
      [theme.breakpoints.up("sm")]: {
        border: "solid 1px",
        borderColor: theme.palette.grey["200"],
        borderRadius: 8,
        boxShadow: "0 4px 12px 1px rgba(208,208,208,0.22)",
        display: "block",
        padding: theme.spacing(2.4),
      },
    },
    "groups-container__content--focus": {
      display: "block",
    },
    "groups-container__content--list": {
      [theme.breakpoints.up("sm")]: {
        maxWidth: 360,
        width: "50%",
      },
    },
    "groups-container__content--main": {
      [theme.breakpoints.up("sm")]: {
        flexGrow: 1,
      },
    },
  };
}
