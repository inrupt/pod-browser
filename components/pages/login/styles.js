/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles } from "@solid/lit-prism-patterns";

// TODO: move these new colors to Prism
const LOGIN_BACKGROUND_COLOR = "#FCFEFF";
const TITLE_COLOR = "#0D6796";

const styles = (theme) =>
  createStyles(theme, ["content"], {
    "login-page": {
      textAlign: "center",
      flexGrow: 1,
      flexShrink: 0,
      backgroundColor: LOGIN_BACKGROUND_COLOR,
      color: theme.palette.grey[800],
      boxSizing: "border-box",
      height: "100%",
      paddingTop: 60, // magic number - same as height of header
      fontStyle: "italic",
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      position: "relative",
      "& p": {
        marginBottom: theme.spacing(3),
      },
    },
    "login-page__title": {
      fontStyle: "normal",
      color: TITLE_COLOR,
      fontSize: "2.3125rem",
      letterSpacing: 1.4,
      lineHeight: "40px",
      margin: theme.spacing(0.7, 0),
    },
    "links-container": {
      fontStyle: "normal",
      padding: theme.spacing(5, 0),
      minWidth: 420,
      maxWidth: 600,
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
    },
    "links-container__text": {
      display: "flex",
      flexDirection: "row",
      justifyContent: "center",
      alignItems: "center",
      marginBottom: theme.spacing(11.6),
    },
    "links-container__list": {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
    },
    "links-container__button": {
      padding: theme.spacing(0, 1),
    },
  });

export default styles;
