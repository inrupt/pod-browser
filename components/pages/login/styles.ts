/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles, PrismTheme } from "@solid/lit-prism-patterns";

const styles = (theme: PrismTheme) =>
  createStyles(theme, ["content"], {
    "login-page": {
      background:
        "linear-gradient(135deg, rgb(124, 77, 255) 0%, rgb(24, 169, 230) 50%, rgb(1, 201, 234) 100%)",
      backgroundRepeat: "no-repeat",
      boxSizing: "border-box",
      height: "100%",
      paddingTop: 60, // magic number - same as height of header
      position: "relative",
      "&::before": {
        content: '""',
        position: "absolute",
        top: 0,
        left: 0,
        width: "100%",
        height: "100%",
        backgroundImage: "url('/background-pattern.svg')",
        filter: "opacity(30%)",
      },
    },
    "login-page__container": {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      position: "relative",
    },
    "login-page__title": {
      color: theme.palette.background.default,
      fontSize: "1.75rem",
      letterSpacing: 1.4,
      lineHeight: "40px",
      margin: theme.spacing(5, 0),
    },
  });

export default styles;
