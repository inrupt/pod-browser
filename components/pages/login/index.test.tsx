import { shallow, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import { ThemeProvider } from "@material-ui/core/styles";
import { useRedirectIfLoggedIn } from "../../../src/effects/auth";
import LoginPage from "./index";

import theme from "../../../src/theme";

jest.mock("../../../src/effects/auth");

describe("Login page", () => {
  test("Renders a logout button", () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <LoginPage />
      </ThemeProvider>
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Redirects if the user is logged out", () => {
    mount(
      <ThemeProvider theme={theme}>
        <LoginPage />
      </ThemeProvider>
    );
    expect(useRedirectIfLoggedIn).toHaveBeenCalled();
  });
});
