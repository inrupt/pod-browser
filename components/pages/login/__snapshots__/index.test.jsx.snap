// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Renders a login button 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-login-page"
  >
    <div
      class="PodBrowser-login-page__container"
    >
      <h1
        class="PodBrowser-content-h1 PodBrowser-login-page__title"
      >
        Hi! Welcome to Solid.
      </h1>
      <div
        class="PodBrowser-login-form"
      >
        <form>
          <div
            class="PodBrowser-root PodBrowser-root"
          >
            <div
              class="PodBrowser-root PodBrowser-root"
            >
              <h3
                data-testid="login-title"
              >
                Log In
              </h3>
              <div
                class="PodBrowser-root"
                style="max-width: 320px; width: 100%;"
              >
                <div
                  aria-expanded="false"
                  class="PodBrowser-root PodBrowser-hasClearIcon"
                  role="combobox"
                >
                  <div
                    class="PodBrowser-root PodBrowser-root PodBrowser-marginNormal PodBrowser-fullWidth"
                    data-testid="login-field"
                  >
                    <label
                      class="PodBrowser-root PodBrowser-root PodBrowser-formControl PodBrowser-animated PodBrowser-outlined"
                      data-shrink="false"
                      for="provider-select"
                      id="provider-select-label"
                    >
                      Select ID provider
                    </label>
                    <div
                      class="PodBrowser-root PodBrowser-root PodBrowser-inputRoot PodBrowser-fullWidth PodBrowser-formControl PodBrowser-adornedEnd PodBrowser-adornedEnd"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="PodBrowser-input PodBrowser-input PodBrowser-input PodBrowser-inputFocused PodBrowser-inputAdornedEnd PodBrowser-inputAdornedEnd"
                        id="provider-select"
                        spellcheck="false"
                        type="url"
                        value=""
                      />
                      <div
                        class="PodBrowser-endAdornment"
                      >
                        <button
                          aria-label="Clear"
                          class="PodBrowser-root PodBrowser-root PodBrowser-clearIndicator"
                          tabindex="-1"
                          title="Clear"
                          type="button"
                        >
                          <span
                            class="PodBrowser-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="PodBrowser-root PodBrowser-fontSizeSmall"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                              />
                            </svg>
                          </span>
                          <span
                            class="PodBrowser-root"
                          />
                        </button>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="PodBrowser-root PodBrowser-notchedOutline"
                      >
                        <legend
                          class="PodBrowser-legendLabelled"
                        >
                          <span>
                            Select ID provider
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                role="button"
                tabindex="0"
              >
                <button
                  class="PodBrowser-button"
                  data-testid="login-button"
                  type="submit"
                >
                  Log In
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
